### PR TITLE
Fix user profile activity feed

### LINKED ISSUE #1202: User's public activity isn't filtered by user
- Gitea version (or commit ref): 1.1.0

## Description

When displaying an user's profile page (https://git.xxxx.io/username?tab=activity), the page shows mixed users activities, not only the current user's activities. 
Is this a normal behavior? 

## Screenshots

hosting

### PR DESCRIPTION
**Pull request by ethantkoenig**
Fixes #1202.

Causes the activity feed on a user's profile page to only be actions performed by that user. The activity feed on the dashboard is unchanged, and still shows relevant actions performed by other users.

### CODE REVIEW

File: models/action.go [hunk 1]
```diff
@@ -672,33 +672,37 @@ func MergePullRequestAction(actUser *User, repo *Repository, pull *Issue) error
...
+	if !options.IncludePrivate {
+		sess.And("is_private = ?", false)
+	}
+	if options.RequestedUser.IsOrganization() {
+		env, err := options.RequestedUser.AccessibleReposEnv(options.RequestingUserID)
 		if err != nil {
 			return nil, fmt.Errorf("AccessibleReposEnv: %v", err)
 		}
-		repoIDs, err := env.RepoIDs(1, ctxUser.NumRepos)
+		repoIDs, err := env.RepoIDs(1, options.RequestedUser.NumRepos)
```
Comment (lunny) [inline models/action.go]: This will make two DB connection in a goroutine. Maybe sqlite will lock table.

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
Otherwise LGTM

File: models/action.go (hunk 1 above)
Comment (ethantkoenig) [reply models/action.go]: I don't see a nice way to use the same connection in both places (i.e. the `sess` variable and this line). If you have a more specific suggestion about how to only make one connection, let me know. 

For what it's worth, the first session (i.e. the `sess` variable) won't actually connect to the database until line 704, so I don't think the two connections will ever overlap.

Comment (lunny) [reply models/action.go]: Maybe change `AccessibleReposEnv` to add a new `Engine` field.

Comment (ethantkoenig) [reply models/action.go]: I've rearranged the code so that these calls take place before `sess` is even created.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Comment by pgaskin** [comment]
LGTM

**Pull request merged**